### PR TITLE
Add ttl for raw client

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: andylokandy <andylokandy@hotmail.com>

- [x] Implementation
- [x] Manual test
- [ ] Add docs

closes #https://github.com/tikv/client-rust/pull/370

### CODE REVIEW

**Comment by jiashiwen** [comment]
This is the feature  I need

File: src/raw/lowering.rs [hunk 1]
```diff
@@ -21,21 +21,30 @@ pub fn new_raw_batch_get_request(
...
 }
 
 pub fn new_raw_batch_put_request(
-    pairs: impl Iterator<Item = KvPair>,
+    pairs_with_ttl: impl Iterator<Item = (KvPair, u64)>,
     cf: Option<ColumnFamily>,
     atomic: bool,
 ) -> kvrpcpb::RawBatchPutRequest {
-    requests::new_raw_batch_put_request(pairs.map(Into::into).collect(), cf, atomic)
+    let (pairs, ttls) = pairs_with_ttl.map(|(pair, ttl)| (pair.into(), ttl)).unzip();
```
Comment (ekexium) [inline src/raw/lowering.rs:46]: Why bother zipping and unzipping?

**Review by ekexium — COMMENTED** [review]
Need some doc comments. Others LGTM.

File: src/raw/lowering.rs (hunk 1 above)
Comment (andylokandy) [reply src/raw/lowering.rs:46]: Do you mean that the API may accept both `pairs` and `ttls`? If so `ttls` and `pairs` can be with different length and we've to add a new error type for it.

Comment (ekexium) [reply src/raw/lowering.rs:46]: The unzipping introduces performance overhead. IMO the method is supposed to be used internally so either returning a string error or just panicking seems fine to me.

**Comment by supercmmetry** [comment]
I really really need this feature!!

Is there a reason why this PR was not merged / worked upon anymore?

**Comment by andylokandy** [comment]
Oh, it was just forgotten.

**Review by iosmanthus — APPROVED** [review]

**Pull request merged**